### PR TITLE
feat: allow fixed size tensors to be used as query vectors

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -60,6 +60,13 @@
         Iterable[RecordBatch],
         pa.RecordBatchReader,
     ]
+    QueryVectorLike = Union[
+        pd.Series,
+        pa.Array,
+        pa.Scalar,
```
Comment (eddyxu) [inline python/python/lance/dataset.py:66]: Why can we use scalar here?

File: python/python/lance/dataset.py [hunk 2]
```diff
@@ -1570,37 +1583,32 @@ def with_fragments(
...
-            isinstance(
-                column_field.type, (pa.FloatingPointArray, np.ndarray, list, tuple)
-            )
-            or pa.types.is_fixed_size_list(column_field.type)
-        ):
+            raise ValueError(f"Embedding column {column} is not in the dataset")
+
+        column_field = self.ds.schema.field(column)
+        column_type = column_field.type
+        if hasattr(column_type, "storage_type"):
```
Comment (eddyxu) [inline python/python/lance/dataset.py:1600]: Do we still need to check storage type for pyarrow > 12?

File: python/python/lance/dataset.py (hunk 1 above)
Comment (westonpace) [reply python/python/lance/dataset.py:66]: It's to handle FixedSizeListScalar (and fixed shape tensor scalar) Pretty often I find myself doing something like:

```
queries = dataset.take(random_indices)
for query in queries:
  # Here `query` is a FixedSizeListScalar
  ...
```

File: python/python/lance/dataset.py (hunk 2 above)
Comment (westonpace) [reply python/python/lance/dataset.py:1600]: Not for fixed shape tensor array.  Once we are using pyarrow 13 we can directly check for `FixedShapeTensorArray`.  However, another advantage of using `storage_type` is that we are future proofing ourselves against other possible extension types.

**Review by eddyxu — APPROVED** [review]

**Pull request merged**